### PR TITLE
i18n(de): Update guides/configuring-astro.md

### PR DESCRIPTION
**Pull request by nbyx**
#### What kind of changes does this PR include?
- New or updated content
- Translated content

#### Description

Add german translation for guides/configuring-astro.md

### CODE REVIEW

File: src/pages/de/guides/configuring-astro.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+## IntelliSense-Konfiguration
+
+Astro empfiehlt die Verwendung der Hilfsfunktion `defineConfig()` in deiner Konfiguration. `defineConfig()` bietet automatisches IntelliSense in deiner IDE. Editoren wie VSCode sind in der Lage Astros Typescript-Typendefinitionen zu lesen und automatische jsdoc-Typ-Hinweise zu liefern, selbst wenn deine Konfiguration nicht in Typescript geschrieben ist.
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  // hier Konfigurationsoptionen einfügen...
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:60]:
```suggestion
  // Füge deine Konfigurationsoptionen hier ein...
```

File: src/pages/de/guides/configuring-astro.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  // https://docs.astro.build/en/reference/configuration-reference/
+})
+```
+
+Du kannst außerdem VSCode auch manuell Typendefinitionen zur Verfügung stellen, indem du diese JSDoc-Notation verwendest:
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:65]:
```suggestion
Du kannst außerdem VS Code auch manuell Typdefinitionen zur Verfügung stellen, indem du diese JSDoc-Notation verwendest: 
```

File: src/pages/de/guides/configuring-astro.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  // https://docs.astro.build/en/reference/configuration-reference/
+})
+```
+
+Du kannst außerdem VSCode auch manuell Typendefinitionen zur Verfügung stellen, indem du diese JSDoc-Notation verwendest: 
+
+```js
+// astro.config.mjs
+export default /** @type {import('astro').AstroUserConfig} */ ({
+  // hier Konfigurationsoptionen einfügen...
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:70]:
```suggestion
  // Füge deine Konfigurationsoptionen hier ein...
```

File: src/pages/de/guides/configuring-astro.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  // https://docs.astro.build/en/reference/configuration-reference/
+}
+```
+
+## Referenzierung relativer Dateien
+
+Wenn du einen relativen Pfad zu `root` oder das `--root`-CLI-flag angibst, wird Astro es gegen das aktuelle Arbeitsverzeichnis auflösen, in dem du den `astro`-CLI-Befehl ausgeführt hast.
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:77]:
```suggestion
Wenn du einen relativen Pfad an `root` oder die Kommandozeilenoption `--root` übergibst, wird Astro diesen relativ zum aktuellen Arbeitsverzeichnis auflösen, in dem du den `astro`-Befehl ausgeführt hast.
```

File: src/pages/de/guides/configuring-astro.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  // Löst in das Verzeichnis "./foo" in dem aktuellen Arbeitsverzeichnis auf
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:84]:
```suggestion
  // Wird ins Verzeichnis "./foo" im aktuellen Arbeitsverzeichnis aufgelöst
```

File: src/pages/de/guides/configuring-astro.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+})
+```
+Astro löst alle anderen relativen Datei- und Verzeichnis-strings relativ zum Projektstammverzeichnis auf:
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  // Löst in das Verzeichnis "./foo" in dem aktuellen Arbeitsverzeichnis auf
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:95]:
```suggestion
  // Wird ins Verzeichnis "./foo" im aktuellen Arbeitsverzeichnis aufgelöst
```

File: src/pages/de/guides/configuring-astro.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+Um eine Datei und ein Verzeichnis relativ zur Konfigurationsdatei zu referenzieren, verwende `import.meta.url` (es sei denn, du schreibst eine common.js-Datei `astro.config.cjs`)
+
+```js "import.meta.url"
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  // Löst das Verzeichnis "./foo" relativ zur Konfigurationsdatei auf
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:109]:
```suggestion
  // Wird ins Verzeichnis "./foo" relativ zu dieser Konfigurationsdatei aufgelöst
```

File: src/pages/de/guides/configuring-astro.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+  root: new URL("./foo", import.meta.url),
+  // Löst das Verzeichnis "./public" relativ zur Konfigurationsdatei auf
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:111]:
```suggestion
  // Wird ins Verzeichnis "./public" relativ zu dieser Konfigurationsdatei aufgelöst
```

File: src/pages/de/guides/configuring-astro.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  },
+})
+```
+
+Dies kann hilfreich sein, wenn du Skripts mit Namen hast, die von Werbeblockern blockiert werden (z.B. `ads.js` oder `google-tag-manager.js`).
+
+
+## Konfigurationsreferenz
+
+📚 In der [API Konfigurationsreferenz](/de/reference/configuration-reference/) von Astro gibt es eine Gesamtübersicht über alle unterstützten Konfigurationsoptionen.
```
Comment (hippotastic) [inline src/pages/de/guides/configuring-astro.md:144]:
```suggestion
📚 In Astros [API-Konfigurationsreferenz](/de/reference/configuration-reference/) findest du eine Gesamtübersicht aller unterstützter Konfigurationsoptionen.
```

**Review by hippotastic — COMMENTED** [review]
Hey, thanks so much for your translation!

I added suggestions for a few stylistic changes and some terminology. Please don't let the number of suggestions distract you from the fact that these are often just the same terms repeating in multiple places. :)

Really solid job overall!

**Comment by nbyx** [comment]
Thanks for your feedback. I changed the translations according to your suggestions. :)

**Comment by hippotastic** [comment]
Thank you for patiently going through all the suggestions! LGTM! :)

**Pull request merged**